clients/dos/src/main/kotlin/DosModel.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.dos
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+
+/**
+ * The JSON (de-)serialization object used by this service.
+ */
+val JSON = Json { ignoreUnknownKeys = true }
+
+@Serializable
+data class UploadUrlRequestBody(
+    val key: String? = null
+)
+
+@Serializable
+data class UploadUrlResponseBody(
+    val success: Boolean,
+    val presignedUrl: String? = null,
+    val message: String? = null
+)
+
+@Serializable
+data class ScanResultsRequestBody(
+    val purls: List<String>? = null,
+    val options: ReqOptions? = null
+) {
+    @Serializable
+    data class ReqOptions(
+        val fetchConcluded: Boolean? = false
+    )
+}
+
+@Serializable
+data class ScanResultsResponseBody(
+    val state: State,
+    val purls: List<String>? = null,
+    val results: JsonObject? = null
+) {
+    @Serializable
+    data class State(
+        val status: String,
+        val jobId: String? = null
+    )
+}
+
+@Serializable
+data class JobRequestBody(
+    val zipFileKey: String? = null,
+    val purls: List<String>? = null
+)
+
+@Serializable
+data class JobResponseBody(
+    val scannerJobId: String? = null,
+    val message: String? = null
+)
+
+@Serializable
+data class JobStateResponseBody(
+    val state: State,
+    val results: JsonObject? = null
+) {
+    @Serializable
+    data class State(
+        val status: String? = null,
+        val message: String? = null
+    )
+}
+
+@Serializable
+data class PackageConfigurationRequestBody(
+    val purl: String? = null
+)
+
+@Serializable
+data class PackageConfigurationResponseBody(
+    val licenseConclusions: List<LicenseConclusion>,
+    val pathExclusions: List<PathExclusion>
+) {
+    @Serializable
+    data class LicenseConclusion(
+        val path: String,
+        val detectedLicenseExpressionSPDX: String? = null,
+        val concludedLicenseExpressionSPDX: String,
+        val comment: String? = null
+    )
```

Comment:
@lamppu or @mmurto can you assist here?

Reply:
I'm not certain what this is.

Reply:
I did look at the DOS code, and as now also mentioned [in my added code comment](https://github.com/oss-review-toolkit/ort/compare/7e641ee5b689eecee1d939e53292fdfdc80c8226..a686d5314f9ac24883809c2589bed5085c4dd599#diff-654c73eb0d695d1a0f435c8e015b9fa1bb44d8d9b820c607b9581f5b88c42ef8R122), this property holds the raw data as reported by ScanCode.

Reply:
In ORT, the semantics of the property is: It selects matching license LicenseFindings with equal detected license, whereas a LicenseFinding is defined as (path, license, start_line, end_line).  

How is it in DOS?

Reply:
I'm again unsure if I get the question correctly. What confuses me is: The *matching* of a license finding curation is always done by ORT core, no matter who the provider of the package configuration was. So how could it be different for DOS if it just provides the package curation, but the matching of a contained license finding curation is done by ORT core?

Reply:
If the semantics of a license finding curation is identical with the one of ORT it could not be different. 
However, I'm trying to understand the semantics of a license finding curation in DOS. I've asked about this several times now, but it remains unanswered. 

The data model is not identical, as it lacks start / end lines, which is why I'm asking this question. It's not obvious at all to me, and documentation pointers I don't have any either.

Reply:
> I've asked about this several times now, but it remains unanswered.

That's not out of ill intent, but simply because I don't understand your questions, and what exactly you're after. Asking for the "semantic of DOS license finding curations" is such a broad question that I have no idea how to answer it pointedly.

So again I'm *guessing* that the concrete question is: Why does DOS not return any `startLines` / `lineCount` as part of a `LicenseFindingCuration`. This is something I don't know and need to check with the team, ping @lamppu.

Reply:
> That's not out of ill intent,

I didn't mean to say it was out of ill intent. 

> Asking for the "semantic of DOS license finding curations" is such a broad question

Hm, I didn't anticipate that this could have been very brought. However, let me try to further narrow down:

Is that detected license property supposed to be matched against

1. Single License findings in ORT `(path, detected,license, start_line, end_line)` or
2. Detected license of the entire file

I do recall some feature flag, which has been introduced not too long ago, which enables some combining to license per file. But I cannot find that commit. However, IIRC it was supposed to be used in DOS context.

Reply:
Yes, so the workflow that we have decided on is that we make curations for the whole file, instead of for individual line matches, so my understanding was that the line numbers could be omitted here. The `detectedLicenseExpressionSPDX` is actually the file-level scanner finding detected by Scancode, as that is what we actually return in our scan results, using the first and last lines of the individual matches as the line numbers. We do save the individual matches in our database, so that curators can inspect these in our UI, but we only return the file-level one to ORT.

Reply:
Thanks @lamppu ! 

So, this means that as-is, ORT would match the license finding curations against single license findings, while it would be required to match them against the license of the entire file. This seems to be a mismatch. Any idea how this could be best addressed, maybe @sschuberth? (Does this curation provider actually require to turn on the feature to make ORT operate on file level licenses instead of single licenses, @sschuberth do you recall your commit for this?)

Reply:
After having another clarifying discussion with @lamppu, the answer is that using the DOS package configuration provider only makes sense when enabling the [`preferFileLicense` configuration](https://github.com/oss-review-toolkit/ort/blob/59681807ea5c4ab9f468e783f2bdb7fa1de5dd30/plugins/scanners/scancode/src/main/kotlin/ScanCode.kt#L59-L65) for ScanCode, or in conjunction with the DOS scanner wrapper. I'll add a `README.md` that documents this.

Reply:
Done. Please have another look @fviernau.

Reply:
Thanks!